lib/collision_detector.cpp

```diff
@@ -0,0 +1,124 @@
+#ifndef FILE_COLLISION_DETECTOR_CPP
+#define FILE_COLLISION_DETECTOR_CPP
+
+#include <limits>
+#include <map>
+#include <unordered_map>
+
+#include "brickengine/collision_detector.hpp"
+#include "brickengine/enum/axis.hpp"
+#include "brickengine/enum/direction.hpp"
+
+#include "brickengine/components/colliders/rectangle_collider_component.hpp"
+#include "brickengine/components/transform_component.hpp"
+
+CollisionDetector::CollisionDetector(std::shared_ptr<EntityManager> em) : entityManager(em) {}
+
+double CollisionDetector::canMove(int entity, Axis axis, Direction direction) {
+    //We only support squares.
+    auto entityRectCollider = entityManager->getComponent<RectangleColliderComponent>(entity);
+    auto entityTransform = entityManager->getComponent<TransformComponent>(entity);
+
+    auto collidableEntities = entityManager->getEntitiesByComponent<RectangleColliderComponent>();
+
+    double spaceLeft = 0; //Distance to the closed object
+    if(direction == Direction::NEGATIVE){
+        spaceLeft = -100000;
+    } else {
+        spaceLeft = 100000;
+    }
+    
+    for(auto& [id, collider] : *collidableEntities) {
+        auto transformComponent = entityManager->getComponent<TransformComponent>(id);
+
+        // int zStartEntity = transformComponent->z;
+        // int zEndEntity = entityRectCollider->z + entityRectCollider->zScale;
+        // int zStartCollidable = transformComponent->z;
+        // int zEndCollidable = transformComponent->z + transformComponent->zScale;
```

Comment:
Remove this?

Reply:
This is future code that will be used for detecting collision on multiple layers. Because of the transform component that is currently not possible but the code needs to be kept